zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala

```diff
@@ -0,0 +1,75 @@
+package zio.kafka.utils
+
+import org.apache.kafka.clients.{ ClientDnsLookup, ClientUtils }
+import zio.{ Task, ZIO }
+
+import java.nio.ByteBuffer
+import java.nio.channels.SocketChannel
+import scala.jdk.CollectionConverters._
+
+/**
+ * This function validates that your Kafka client (Admin, Consumer, or Producer) configurations are valid for the Kafka
+ * Cluster you want to contact.
+ *
+ * This function protects you against this long standing bug in kafka-clients that leads to crash your app with an OOM.
+ * More details, see: https://issues.apache.org/jira/browse/KAFKA-4090
+ */
+object SslHelper {
+  def validateEndpoint(bootstrapServers: List[String], props: Map[String, AnyRef]): Task[Unit] =
+    ZIO
+      .unless(
```

Comment:
🌕 I'd prefer we use `.when`. It feels like a double negation which is hard to understand

Reply:
unless is much simpler in this case, because we need to check all cases except for SSL

Reply:
Ok I see